src/utils/network/core.cljs

```diff
@@ -41,7 +41,12 @@
        (map :error)
        (not-any? identity)))
 
-(defn chain-id-available?
-  [current-networks network]
-  (let [chain-id (get-in network [:config :NetworkId])]
-    (every? #(not= chain-id (get-in % [1 :config :NetworkId])) current-networks)))
+(defn fetch [url {:keys [body] :as params} callback]
+  (let [js-params (cond-> params
+                    (map? body) (update :body (comp js/JSON.stringify clj->js))
+                    :always     clj->js)]
+    (-> (js/fetch url js-params)
+        (.then (fn [response]
+                 (.text response)))
+        (.then callback)
+        (.catch #(log/error (str "Error while fetching: " url) %)))))
```

Comment:
Added a general `fetch` function that is wrapping `js/fetch` to use it in a Clojure friendly way

Reply:
I always find confusing to mix promises with callbacks. The `fetch` function is returning a promise, so we don't need a callback. I've only seen this style of code mixing promises with callbacks in status-mobile.

But I imagine you did it like this to avoid further changes in the repo? Which would be caused if you used an implementation like the one below.

```clojure
(defn fetch [url {:keys [body] :as params}]
  (let [js-params (cond-> params
                    (map? body) (update :body (comp js/JSON.stringify clj->js))
                    :always     clj->js)]
    (-> (js/fetch url js-params)
        (.then (fn [response]
                 (.text response)))
        (.catch #(log/error (str "Error while fetching: " url) %)))))
```

I think it's best to keep changes at a minimum, but I wanted to point out @ulisesmac. Sorry if you already know all this.

Reply:
Thanks for paying attention to this @ilmotta 

I even considered to return `nil` for this function, so it could be considered just as a function with side effects.

I know it's confusing, but I wanted to pass a callback as a way to easily deal with `fetch`, not having to worry about dealing with a promise, possibbly import  `promesa.core` and propagate the promse (e.g. `status-backend.core/fetch` would also return a promise).

IMO we don't need all the flexibility/power of a promise for `js/fetch`, we just care about having a callback (or two: on-success and on-failure) to receive the result.

I'm open to listening to opinions on why this approach is worse. I wanted to keep things simple and familiar.